### PR TITLE
fix sidecarset ExpectUpdated block upgrade container

### PR DESCRIPTION
**Pull request by zmberg**

### CODE REVIEW

File: pkg/control/sidecarcontrol/sidecarset_control.go [hunk 1]
```diff
@@ -181,12 +181,12 @@ func (c *commonControl) IsPodStateConsistent(pod *v1.Pod, sidecarContainers sets
 	return IsSidecarContainerUpdateCompleted(pod, sets.NewString(sidecarset.Name), sidecarContainers)
 }
 
-// k8s only allow modify pod.spec.container[x].image,
-// only when annotations[SidecarSetHashWithoutImageAnnotation] is the same, sidecarSet can upgrade pods
-func (c *commonControl) IsSidecarSetUpgradable(pod *v1.Pod) bool {
+func (c *commonControl) IsSidecarSetUpgradable(pod *v1.Pod) (canUpgrade, consistent bool) {
```
Comment (furykerry) [inline pkg/control/sidecarcontrol/sidecarset_control.go:184]: plz comment about what consistent means

File: pkg/controller/sidecarset/sidecarset_processor.go [hunk 2]
```diff
@@ -163,7 +163,8 @@ func (p *Processor) updatePods(control sidecarcontrol.SidecarControl, pods []*co
 			klog.Errorf("update NotUpgradable PodCondition error, s:%s, pod:%s, err:%v", sidecarset.Name, pod.Name, err)
 			return err
 		}
-		sidecarcontrol.UpdateExpectations.ExpectUpdated(sidecarset.Name, sidecarcontrol.GetSidecarSetRevision(sidecarset), pod)
+		// Since the pod sidecarSet hash is not updated here, it cannot be called ExpectUpdated
```
Comment (furykerry) [inline pkg/controller/sidecarset/sidecarset_processor.go:166]: just remove the code? we only comment code that will be uncommented in the future

Comment (veophi) [reply pkg/controller/sidecarset/sidecarset_processor.go:166]: // TODO: add ResourceVersionExpectation instead of UpdateExpectations

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**